ui/desktop/electron-app/src/index.js

```diff
@@ -184,33 +108,118 @@ app.on('ready', async () => {
 
   // Prevent navigation outside of serve://boundary per
   // Electronegativity LIMIT_NAVIGATION_GLOBAL_CHECK
-  mainWindow.webContents.on('will-navigate', (event, url) => {
+  browserWindow.webContents.on('will-navigate', (event, url) => {
     /* eng-disable LIMIT_NAVIGATION_JS_CHECK */
     if (!url.startsWith('serve://boundary')) event.preventDefault();
   });
+
   // Opens external links in the host default browser.
   // We just allow boundaryproject.io domain to open on external window (for now).
-  mainWindow.webContents.on('new-window', (event, url) => {
+  browserWindow.webContents.on('new-window', (event, url) => {
     /* eng-disable LIMIT_NAVIGATION_JS_CHECK */
     event.preventDefault();
     if (url.startsWith('https://boundaryproject.io/')) {
       shell.openExternal(url);
     }
   });
 
-  mainWindow.on('unresponsive', () => {
+  browserWindow.on('unresponsive', () => {
     console.log(
       'Your Ember app (or other code) has made the window unresponsive.'
     );
   });
 
-  mainWindow.on('responsive', () => {
+  browserWindow.on('responsive', () => {
     console.log('The main window has become responsive again.');
   });
 
-  mainWindow.on('closed', () => {
-    mainWindow = null;
+  browserWindow.on('closed', () => {
+    closeWindowCB();
+  });
+
+  return browserWindow;
+};
+
+let mainWindow = null;
```

Comment:
Do we still need to keep and set this variable? It looks like we don't do anything with this afterwards after creating the window. If we remove it, we don't have to return the `browserWindow` from `createWindow` and the `closed` event can still be as it was before. 

I think this still encapsulates setting the event handlers for the main window, though if the goal is to have this be a pure function and return the window even if we don't do anything with it currently, I'm fine with this.

Reply:
Hey @ZedLi 

This is declaration and initialization of `mainWindow` variable, we set it to null on initialization and when we destroy the window. So it is not undefined.

The purpose of having `mainWindow` is that we can re-create the window when handling event `active`. More details about this in this PR.

That said, happy to discuss your approach over a zoom call? Not sure I am getting 100% of it and interested on hear more about it :smiley:

Reply:
Had a quick conversation, I forgot that the objective was to fix the mac os window lifecycle and the other PR uses the `mainWindow`, makes sense to me!